pkgs/applications/window-managers/weston/default.nix

```diff
@@ -1,44 +1,43 @@
-{ stdenv, fetchurl, pkgconfig, wayland, libGL, mesa_noglu, libxkbcommon, cairo, libxcb
-, libXcursor, xlibsWrapper, udev, libdrm, mtdev, libjpeg, pam, dbus, libinput
+{ stdenv, fetchurl, meson, ninja, pkgconfig, wayland, libGL, mesa_noglu, libxkbcommon, cairo, libxcb
+, libXcursor, xlibsWrapper, udev, libdrm, mtdev, libjpeg, pam, dbus, libinput, libevdev
+, colord, lcms2
 , pango ? null, libunwind ? null, freerdp ? null, vaapi ? null, libva ? null
 , libwebp ? null, xwayland ? null, wayland-protocols
 # beware of null defaults, as the parameters *are* supplied by callPackage by default
 }:
 
+with stdenv.lib;
 stdenv.mkDerivation rec {
   name = "weston-${version}";
-  version = "5.0.0";
+  version = "6.0.0";
 
   src = fetchurl {
     url = "https://wayland.freedesktop.org/releases/${name}.tar.xz";
-    sha256 = "1bsc9ry566mpk6fdwkqpvwq2j7m79d9cvh7d3lgf6igsphik98hm";
+    sha256 = "04p6hal5kalmdp5dxwh2h5qhkkb4dvbsk7l091zvvcq70slj6qsl";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ meson ninja pkgconfig ];
   buildInputs = [
     wayland libGL mesa_noglu libxkbcommon cairo libxcb libXcursor xlibsWrapper udev libdrm
-    mtdev libjpeg pam dbus libinput pango libunwind freerdp vaapi libva
+    mtdev libjpeg pam dbus libinput libevdev pango libunwind freerdp vaapi libva
     libwebp wayland-protocols
+    colord lcms2
   ];
 
-  configureFlags = [
-    "--enable-x11-compositor"
-    "--enable-drm-compositor"
-    "--enable-wayland-compositor"
-    "--enable-headless-compositor"
-    "--enable-fbdev-compositor"
-    "--enable-screen-sharing"
-    "--enable-clients"
-    "--enable-weston-launch"
-    "--disable-setuid-install" # prevent install target to chown root weston-launch, which fails
-  ] ++ stdenv.lib.optional (freerdp != null) "--enable-rdp-compositor"
-    ++ stdenv.lib.optional (vaapi != null) "--enable-vaapi-recorder"
-    ++ stdenv.lib.optionals (xwayland != null) [
-        "--enable-xwayland"
-        "--with-xserver-path=${xwayland.out}/bin/Xwayland"
-      ];
+  mesonFlags= [
+    "-Dbackend-rdp=${boolToString (freerdp != null)}"
+    "-Dremoting=false" # TODO
+    "-Dsimple-dmabuf-drm=" # Disables all drivers
+    "-Dtest-junit-xml=false"
+    #"--enable-clients"
```

Comment:
This was introduced here https://gitlab.freedesktop.org/wayland/weston/commit/60c15080ab24ae2f4347fb04e39f147c51656688 and the clients are defined here: https://gitlab.freedesktop.org/wayland/weston/blob/6.0.0/Makefile.am#L565